src/Configuration.php

```diff
@@ -0,0 +1,198 @@
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+declare(strict_types=1);
+
+namespace Doctrine\Annotations;
+
+use Doctrine\Annotations\Metadata\MetadataFactory;
+use Doctrine\Annotations\Parser\MetadataParser;
+use Doctrine\Annotations\Parser\HoaParser;
+use Doctrine\Annotations\Parser\PhpParser;
+
+/**
+ * Annotation parser configuration.
+ *
+ * @author Fabio B. Silva <fabio.bat.silva@gmail.com>
+ */
+class Configuration
+{
+    /**
+     * A list with annotations that are not causing exceptions when not resolved to an annotation class.
+     *
+     * The names must be the raw names as used in the class, not the fully qualified
+     * class names.
+     *
+     * @var \Doctrine\Annotations\IgnoredAnnotationNames
+     */
+    private $ignoredAnnotationNames;
+
+    /**
+     * @var \Doctrine\Annotations\Metadata\MetadataFactory
+     */
+    private $metadataFactory;
+
+    /**
+     * @var \Doctrine\Annotations\Parser\PhpParser
+     */
+    private $phpParser;
+
+    /**
+     * @var \Doctrine\Annotations\Parser\HoaParser
+     */
+    private $hoaParser;
+
+    /**
+     * @var \Doctrine\Annotations\Resolver
+     */
+    private $resolver;
+
+    /**
+     * @var \Doctrine\Annotations\Builder
+     */
+    private $builder;
+
+    /**
+     * @param \Doctrine\Annotations\IgnoredAnnotationNames $names
+     */
+    public function setIgnoredAnnotationNames(IgnoredAnnotationNames $names)
+    {
+        $this->ignoredAnnotationNames = $names;
+    }
+
+    /**
+     * @return \Doctrine\Annotations\IgnoredAnnotationNames
+     */
+    public function getIgnoredAnnotationNames() : IgnoredAnnotationNames
+    {
+        if ($this->ignoredAnnotationNames !== null) {
+            return $this->ignoredAnnotationNames;
+        }
+
+        return $this->ignoredAnnotationNames = new IgnoredAnnotationNames(IgnoredAnnotationNames::DEFAULT_NAMES);
+    }
+
+    /**
+     * @param \Doctrine\Annotations\Metadata\MetadataFactory $factory
+     */
+    public function setMetadataFactory(MetadataFactory $factory)
+    {
+        $this->metadataFactory = $factory;
+    }
+
+    /**
+     * @return \Doctrine\Annotations\Metadata\MetadataFactory
```

Comment:
Sometimes you're using FQCN and sometimes you're using alias. It'd be better to use only one.